### ISSUE TITLE
More explanation on errors please

### ISSUE THREAD

**Issue by carinlynchin** [ariya/phantomjs#14848]
1. Which version of PhantomJS are you using? Tip: run `phantomjs --version`.
      2.1.1

2. What steps will reproduce the problem?
  1. just running a js
  2.
  3.

3. Which operating system are you using?
3.13.0-86-generic ubuntu

4. Did you use binary PhantomJS or did you compile it from source?
binary

5. Please provide any additional information below.
getting this error 
SyntaxError: Expected an identifier but found 'not' instead

not sure where it comes from but every time i get an error there is no list of explanations... sometimes i have to dig in source code here just to try and figure out what caused it

**Comment by Vitallium** [comment]
Could you please post your JavaScript code?

**Comment by carinlynchin** [comment]
I can't right now, but really this issue I created was to have some kind of documentation showing what the errors mean. That way its easier to try and track down where the issue is. I have used Debug but it only gets me so far... for example this latest error is 


evaluateJavaScript result QVariant(Invalid)


i don't even understand what the error means, so how can I understand what caused it?  You know what i mean?
